src/engine/bank05.asm

```diff
@@ -54,31 +54,1059 @@ PointerTable_14000: ; 14000 (05:4000)
 	dw $48dc ; IMAKUNI_DECK
 ; 1406a
 
-	INCROM $1406a, $14226
+PointerTable_1406a: ; 1406a (5:406a)
+	dw $406c
+	dw Func_14078
+	dw Func_14078
+	dw $409e
+	dw $40a2
+	dw $40a6
+	dw $40aa
+
+Func_14078: ; 14078 (5:4078)
+	call Func_15eae
+	call Func_158b2
+	jr nc, .asm_14091
+	call Func_15b72
+	call Func_15d4f
+	call Func_158b2
+	jr nc, .asm_14091
+	call Func_15b72
+	call Func_15d4f
+.asm_14091
+	call Func_164e8
+	call Func_169f8
+	ret c
+	ld a, $05
+	bank1call AIMakeDecision
+	ret
+; 0x1409e
+
+	INCROM $1409e, $140ae
+
+; returns carry if damage dealt from any of
+; a card's moves KOs defending Pokémon
+; outputs index of the move that KOs
+; input:
+; 	[hTempPlayAreaLocation_ff9d] = location of attacking card to consider
+; output:
+; 	[wSelectedMoveIndex] = move index that KOs
+CheckIfAnyMoveKnocksOutDefendingCard: ; 140ae (5:40ae)
+	xor a ; first move
+	call CheckIfMoveKnocksOutDefendingCard
+	ret c
+	ld a, $01 ; second move
+;	fallthrough
+
+CheckIfMoveKnocksOutDefendingCard: ; 140b5 (5:40b5)
+	call CalculateMoveDamage_VersusDefendingCard
+	ld a, DUELVARS_ARENA_CARD_HP
+	call GetNonTurnDuelistVariable
+	ld hl, wDamage
+	sub [hl]
+	ret c
+	ret nz
+	scf
+	ret
+; 0x140c5
+
+	INCROM $140c5, $140df
+
+; checks AI scores for all benched Pokémon
+; returns the location of the card with highest score
+; in hTempPlayAreaLocation_ff9d
+FindHighestBenchScore: ; 140df (5:40df)
+	ld a, DUELVARS_NUMBER_OF_POKEMON_IN_PLAY_AREA
+	call GetTurnDuelistVariable
+	ld b, a
+	ld c, 0
+	ld e, c
+	ld d, c
+	ld hl, wBenchAIScore + 1
+	jp .next
+
+.loop
+	ld a, [hli]
+	cp e
+	jr c, .next
+	ld e, a
+	ld d, c
+.next
+	inc c
+	dec b
+	jr nz, .loop
+
+	ld a, d
+	ldh [hTempPlayAreaLocation_ff9d], a
+	or a
+	ret
+; 0x140fe
+
+; adds a to wAIScore
+; if there's overflow, it's capped at $ff
+; output:
+; 	a = a + wAIScore (capped at $ff)
+AddToAIScore: ; 140fe (5:40fe)
+	push hl
+	ld hl, wAIScore
+	add [hl]
+	jr nc, .no_cap
+	ld a, $ff
+.no_cap
+	ld [hl], a
+	pop hl
+	ret
+; 0x1410a
+
+; subs a from wAIScore
+; if there's underflow, it's capped at $00
+SubFromAIScore: ; 1410a (5:410a)
+	push hl
+	push de
+	ld e, a
+	ld hl, wAIScore
+	ld a, [hl]
+	or a
+	jr z, .done
+	sub e
+	ld [hl], a
+	jr nc, .done
+	ld [hl], $00
+.done
+	pop de
+	pop hl
+	ret
+; 0x1411d
+
+; stores defending Pokémon's weakness/resistance
+; and the number of prize cards in both sides
+StoreDefendingPokemonColorWRAndPrizeCards: ; 1411d (5:411d)
```

Comment:
Just semantics would I would use Load over Store here. I think it better suggests that some card properties are loaded into RAM for subsequent use. Store sounds more like for something that will stay there for a long time rather than for temporary calculations, like data written to SRAM.